### PR TITLE
Added project euler problem 80

### PR DESCRIPTION
**Pull request by Secozzi**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_80/sol1.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+https://projecteuler.net/problem=80
+
+It is well known that if the square root of a natural number is not an integer,
+then it is irrational. The decimal expansion of such square roots is infinite
+without any repeating pattern at all.
+
+The square root of two is 1.41421356237309504880...,
+and the digital sum of the first one hundred decimal digits is 475.
+
+For the first one hundred natural numbers, find the total of the digitalcls
```
[the remark this answers is not in the document]
Comment (Secozzi) [reply project_euler/problem_80/sol1.py]: Fixed now.

**Review by ruppysuppy — APPROVED** [review]

File: project_euler/problem_80/sol1.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    >>> sqrt_with_accuracy(80, 100)
+    '8944271909999158785636694674925104941762473438446102897083588981642083702551219597657657633515129099'
+    """
+    getcontext().prec = accuracy + 5
+    ans = str(Decimal(num).sqrt())
+    ans = ans.replace(".", "")
+    return ans[:accuracy]
+
+
+def solution():
```
Comment (cclauss) [inline project_euler/problem_80/sol1.py:34]: Please break this into two functions.  The first should return the value for just one number and should have a doctest that proves the answer on lines 8 and 9.  The second should use the first to calculate the numbers between 2 and 100.

Is there a source that tells us that 40886 is the correct answer?

**Review by cclauss — CHANGES_REQUESTED** [review]

File: project_euler/problem_80/sol1.py (hunk 2 above)
Comment (Secozzi) [reply project_euler/problem_80/sol1.py:41]: I've split the functions, but I'm unsure where to find a source other than entering it into https://projecteuler.net/problem=80 and showing that 40886 is indeed the answer.

File: project_euler/problem_80/sol1.py [hunk 3]
```diff
@@ -16,39 +16,45 @@
...
+        total_sum += int(digit)
+
+    return total_sum
 
 
 def solution():
     """Returns the total of the digital sum of the first one hundred
     decimal digits for all the irrational square roots of all the first
-    one hundred numbers.
+    one hundrer numbers.
```
Comment (cclauss) [inline project_euler/problem_80/sol1.py:44]: ?

File: project_euler/problem_80/sol1.py (hunk 2 above)
Comment (cclauss) [reply project_euler/problem_80/sol1.py:41]: Why get rid of `sqrt_with_accuracy()`?  That is a cool function and is useful in other contexts.  Let’s bring that one back and we can feed it’s output into a much simpler `sum_of_digits()’.

File: project_euler/problem_80/sol1.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    ans = str(Decimal(num).sqrt())
+    return ans[: accuracy + 1]
+
+
+def digital_sum(num: str) -> int:
+    """Function to calculate the digital sum of a number.
+    The digital sum is the sum of the digits of the representation of a number
+    Returns the digital sum as an int.
+
+    >>> digital_sum("1.414213562373095048801688724209698078569" \
```
Comment (cclauss) [inline project_euler/problem_80/sol1.py:36]: \ is not needed inside () or [] or {}.  \ line continuation is strongly discouraged in PEP8 and other places.

File: project_euler/problem_80/sol1.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    numbers = [num for num in numbers if num not in (1, 4, 9, 16, 25, 36, 49, 64, 81)]
+
+    total_sum = 0
+    accuracy = 100
+
+    for num in numbers:
+        sqrt_num = sqrt_with_accuracy(num, accuracy)
+        total_sum += digital_sum(sqrt_num)
+
+    return total_sum
```
Comment (cclauss) [inline project_euler/problem_80/sol1.py:69]: Please use the Python builtin `sum()` function with a list comprehension/ generator expression to make this function a one liner.

Comment (spamegg1) [reply project_euler/problem_80/sol1.py:69]: @ruppysuppy You can simplify `digital_sum` to:
```python
def digital_sum(num: str) -> int:
    # docstring...
    # doctests...
    return sum(int(digit) for digit in num.replace(".", ""))
```
and you can simplify `solution` to:
```python
def solution():
    # docstring...
    # doctests...
squares = [i ** 2 for i in range(1, 10)]
numbers = [num for num in range(1, 100) if num not in squares]
return sum(digital_sum(sqrt_with_accuracy(num, 100)) for num in numbers)
```

**Pull request closed without merging**